### PR TITLE
fine-tUME-1

### PR DESCRIPTION
**Pull request by taylormjs**
## Description:
This PR introduces supervised fine-tuning for base UME models. Feature additions include:
- Task heads (regression & classification), configs and loss functions to enable finetuning UME on dowstream tasks
- cmdline.finetune module for easy finetuning with CLI
- Unfreezing module to unfreeze n layers of the encoder for fine-tuning
- Loss fn and activation fn registries
- UMEPropertyRegression to easily wrap everything for hydra-compatible experiments
- New post_train directory for SFT+ modules to live
- Small bug fix, including using torchmetrics in the old ppi_clf module (likely to be deprecated)
- Added ume-large-base-870M as additional model due to issues loading in large-base-740M

## Type of Change
- [x] Bug fix
- [x] New feature  
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring

### CODE REVIEW

File: src/lobster/model/losses/_regression.py [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+        Moment averaging label smoothing: Cortex (https://github.com/prescient-design/cortex/blob/main/cortex/model/leaf/_regressor_leaf.py)
+        Moment averaging theory: "Adaptive Approximate Inference in Bayesian Neural Networks" (Maddison et al., 2017)
+        https://www.cs.toronto.edu/~cmaddis/pubs/aais.pdf
+        """
+        # Compute standard statistics
+        pred_mean = pred.mean()
+        pred.var()
+
+        # Apply smoothing between prediction and target statistics
+        smoothed_target = self.label_smoothing * pred_mean + (1.0 - self.label_smoothing) * target
```
Comment (jks17) [inline src/lobster/model/losses/_regression.py:92]: Has this been shown to be useful in Cortex? It looks like we remove the cross-var terms that were in the original code - is this an issue? I also don't see how this is useful unless the label_smoothing term alters during training but maybe I am misunderstanding the paper quoted.

File: pyproject.toml [hunk 2]
```diff
@@ -4,10 +4,10 @@ readme = "README.md"
 description = "Language models for Biological Sequence Transformation and Evolutionary Representation."
 authors = [{name = "Nathan C. Frey", email = "frey.nathan.nf1@gene.com"}]
 dynamic = ["version"]
-requires-python = ">=3.11, <3.13"
+requires-python = ">=3.11, <3.14"
 dependencies = [
     "pooch",
-    "torch",
```
Comment (karinazad) [inline pyproject.toml:10]: is the version pin needed?

File: src/lobster/model/_heads.py [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+            raise ValueError("Binary classification should have output_dim=1")
+
+        # Handle auto activation selection
+        if self.activation == "auto":
+            self.activation = get_recommended_activation(self.task_type)
+
+        # Validate activation function
+        if self.activation.lower() not in ACTIVATION_FUNCTIONS and self.activation.lower() != "swiglu":
+            available = list(ACTIVATION_FUNCTIONS.keys())
+            raise ValueError(f"Unknown activation function: {self.activation}. Available: {available}")
```
Comment (karinazad) [inline src/lobster/model/_heads.py:85]: validation could be handled sindie `get_activation_function` so we don't need the `ACTIVATION_FUNCTIONS` dict

File: src/lobster/model/_heads.py [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+        self,
+        input_dim: int,
+        task_config: TaskConfig,
+        encoder_config: Any | None = None,
+    ):
+        super().__init__()
+        self.task_config = task_config
+
+        # Create pooler using existing infrastructure
+        pooler_class = POOLERS[task_config.pooling]
```
Comment (karinazad) [inline src/lobster/model/_heads.py:118]: I think it would be good to standardize when we call helpers (`get_activation, get_pooler` etc) and when we access dicts directly. I think helpers are nicer?

File: src/lobster/model/_ppi_clf.py [hunk 5]
```diff
@@ -5,8 +5,8 @@
 import torch
 import torch.nn as nn
 from torch.nn import BCEWithLogitsLoss
-from torcheval.metrics import BinaryAUPRC
```
Comment (karinazad) [inline src/lobster/model/_ppi_clf.py:8]: nice

File: src/lobster/model/_ume_property_regression.py [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+import logging
+from typing import Literal
+
+import lightning as L
+from torchmetrics import MeanAbsoluteError, R2Score, SpearmanCorrCoef, PearsonCorrCoef
+import torch
+
+from ._heads import TaskConfig, FlexibleEncoderWithHeads
+from ._ume import UME
+from ..post_train.unfreezing import set_unfrozen_layers
```
Comment (karinazad) [inline src/lobster/model/_ume_property_regression.py:11]: I'm a bit worried about circular dependencies here but it looks fine

maybe just do `from lobster.post_train.unfreezing ...` for consistency?

File: src/lobster/post_train/unfreezing.py [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+                "lr": base_lr,
+                "layer_id": "other",
+            }
+        )
+
+    logger.info(f"Created {len(param_groups)} parameter groups with layer-wise learning rates")
+    return param_groups
+
+
+def progressive_unfreezing_schedule(model: UME, current_epoch: int, unfreeze_schedule: list[int]) -> None:
```
Comment (karinazad) [inline src/lobster/post_train/unfreezing.py:202]: ditto

File: src/lobster/model/losses/_regression.py (hunk 1 above)
Comment (taylormjs) [reply src/lobster/model/losses/_regression.py:92]: Yeah I think you're right. I'll go ahead and remove it from MSELoss. I think the full version with cross terms can just be added to the NaturalGaussianLoss, so I'll add there here or a follow-up PR
